spec/models/metric_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+RSpec.describe Metric, type: :model do
+  subject { build :metric }
+
+  describe 'database schema' do
+    it 'has an entity_key field' do
+      expect(subject).to have_db_column(:entity_key)
+        .of_type(:string)
+        .with_options(null: false)
+    end
+
+    it 'has an metric_key field' do
+      expect(subject).to have_db_column(:metric_key)
+        .of_type(:string)
+        .with_options(null: false)
+    end
+
+    it 'has a value field' do
+      expect(subject).to have_db_column(:value)
+        .of_type(:decimal)
+        .with_options(null: true)
+    end
+
+    it 'has a value_timestamp field' do
+      expect(subject).to have_db_column(:value_timestamp)
+        .of_type(:datetime)
+        .with_options(null: true)
+    end
+  end
+
+  describe 'validation' do
+    describe 'of entity_key' do
+      it 'fails with a missing value' do
+        subject.entity_key = nil
+
+        expect(subject).to_not be_valid
```

Comment:
doesn't `validate_presence_of` work for you?

Reply:
Did not know about it. Will try it tomorrow. Thanks!

Reply:
Done. Thanks!